dali/imgcodec/decoders/nvjpeg/nvjpeg.cc

```diff
@@ -144,6 +144,19 @@ NvJpegDecoderInstance::PerThreadResources::~PerThreadResources() {
   }
 }
 
+bool NvJpegDecoderInstance::CanDecode(DecodeContext ctx, ImageSource *in, DecodeParams opts,
```

Comment:
By implementing CanDecode, we can detect lossless JPEGs (not supported by this backend), and fail earlier